scripts/db.dropcolumn/db.dropcolumn.py

```diff
@@ -80,30 +80,38 @@ def main():
         return 0
 
     if driver == "sqlite":
-        # echo "Using special trick for SQLite"
-        # http://www.sqlite.org/faq.html#q13
-        colnames = []
-        coltypes = []
-        for f in gscript.db_describe(table)["cols"]:
-            if f[0] != column:
-                colnames.append(f[0])
-                coltypes.append("%s %s" % (f[0], f[1]))
-
-        colnames = ", ".join(colnames)
-        coltypes = ", ".join(coltypes)
-
-        cmds = [
-            "BEGIN TRANSACTION",
-            "CREATE TEMPORARY TABLE ${table}_backup(${coldef})",
-            "INSERT INTO ${table}_backup SELECT ${colnames} FROM ${table}",
-            "DROP TABLE ${table}",
-            "CREATE TABLE ${table}(${coldef})",
-            "INSERT INTO ${table} SELECT ${colnames} FROM ${table}_backup",
-            "DROP TABLE ${table}_backup",
-            "COMMIT",
-        ]
-        tmpl = string.Template(";\n".join(cmds))
-        sql = tmpl.substitute(table=table, coldef=coltypes, colnames=colnames)
+        sqlite3_version = gscript.read_command("sqlite3", "version")
```

Comment:
This would require the SQLite binaries to be installed.

Perhaps rather like here:
`python3 -c "import sqlite3; print(sqlite3.sqlite_version)"`
?

Reply:
Or perhaps `SELECT sqlite_version();`?

Reply:
> Or perhaps `SELECT sqlite_version();`?

I really like that. I changed it.

Reply:
> > Or perhaps `SELECT sqlite_version();`?
> 
> I really like that. I changed it.

You should probably also guard for older version that do not have the `sqlite_version()` function (https://www.techonthenet.com/sqlite/functions/sqlite_version.php).

Reply:
Probably nothing to worry about : https://github.com/sqlite/sqlite/commit/647cb0e159cfe8a42691bd29a8e1820efa1a171c

Reply:
21 years - I would disregard this.